backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/switch/BpkSwitch.kt

```diff
@@ -73,15 +79,25 @@ fun BpkSwitch(
     Row(
         verticalAlignment = Alignment.CenterVertically,
         horizontalArrangement = Arrangement.SpaceBetween,
-        modifier = modifier.applyIf(onCheckedChange != null) {
-            toggleable(
-                value = checked,
-                role = Role.Switch,
-                interactionSource = interactionSource,
-                indication = null,
-                onValueChange = onCheckedChange!!,
-            )
-        },
+        modifier = modifier
+            .applyIf(onCheckedChange != null) {
+                toggleable(
+                    value = checked,
+                    role = Role.Switch,
+                    interactionSource = interactionSource,
+                    indication = null,
+                    onValueChange = onCheckedChange!!,
+                    enabled = enabled,
+                )
+            }.applyIf(onCheckedChange == null) {
+                semantics(mergeDescendants = true) {
+                    role = Role.Switch
+                    if (!enabled) {
+                        disabled()
+                        toggleableState = ToggleableState(checked)
```

Comment:
can you move this out of the `enabled` if statement? that way it'll also be announced for an enabled switch if there's no click listener (not that that should ever be done...)